src/ui/common/src/components/integrations/dialogs/snowflakeDialog.tsx

```diff
@@ -16,27 +14,11 @@ const Placeholders: SnowflakeConfig = {
 };
 
 type Props = {
-  setDialogConfig: (config: IntegrationConfig) => void;
+  onUpdateField: (field: keyof SnowflakeConfig, value: Stringifiable) => void;
+  value?: SnowflakeConfig;
 };
 
-export const SnowflakeDialog: React.FC<Props> = ({ setDialogConfig }) => {
-  const [accountIdentifier, setAccountIdentifier] = useState<string>(null);
```

Comment:
@agiron123 lmk if this make sense to you. Basically we directly pass external state as 'value' and use `onUpdateField` to set the external state, and in this way this component becomes stateless and we removed the original state copying

Reply:
Makes sense to me. Nice react pattern of "lifting state up" being used here :)